docs/src/description/buses.md

```diff
@@ -0,0 +1,61 @@
+# Buses
+
+A bus is a construct which aims to simplify description of non-local constraints defined via multiset or LogUp checks.
+
+## Bus types
+
+- Multiset (`unit`): Multiset-based buses can represent constraints specifying given values have been added or removed from a column, in no specific order.
+[Citation needed]()
+
+- LogUp (`mult`): LogUp-based buses are more complex than multiset buses, and can encode the multiplicity of an element: an element can be added or removed multiple times.
+[Multivariate lookups based on logarithmic derivatives - Ulrich Haböck, Orbis Labs, Polygon Labs](https://eprint.iacr.org/2022/1530)
+
+## Defining buses
+
+See the [declaring buses](./declarations.md#buses) for more details.
+
+```
+buses {
+    unit p,
+    mult q,
+}
+```
+
+## Bus boundary constraints
+
+In the boundary constraints section, we can constrain the initial and final state of the bus. Currently, only constraining a bus to be empty (with the  `null` keyword) is supported.
+
+```
+boundary_constraints {
+    enf p.first = null;
+    enf p.last = null;
+}
+```
+
+The above example states that the bus `p` should be empty at the beginning and end of the trace.
+
+## Bus integrity constraints
+
+In the integrity constraints section, we can add and remove elements (as tuples of felts) to and from a bus. In the following examples, `p` and `q` are respectivelly multiset and LogUp based buses.
+
+```
+integrity_constraints {
+    p.add(a) when s1;
+    p.rem(a, b) when 1 - s2;
```

Comment:
I feel like `add` and (particularly) `rem` visually just look too much like their usual mathematical operations, rather than something more set-oriented. Maybe that's just my misconception in terms of understanding what is implied by these operations, but Is there any reason not to use `insert`/`append` and `remove`/`delete` for these?

Generally speaking, I really prefer _not_ abbreviating things unless it is absolutely unambiguous what the name refers to. In this case, I don't think that's true, and more explicit language is likely to make the resulting AirScript more readable. I feel the same way about `enf` versus `enforce`, but never got around to changing that one myself.

Anyway, thoughts? /cc @bobbinth